qbft/round_change.go

```diff
@@ -227,12 +228,12 @@ func validRoundChangeForData(
 		return errors.New("msg allows 1 signer")
 	}
 
-	if err := signedMsg.Signature.VerifyByOperators(signedMsg, config.GetSignatureDomainType(), types.QBFTSignatureType, state.Share.Committee); err != nil {
-		return errors.Wrap(err, "msg signature invalid")
+	if err := signedMsg.Validate(); err != nil {
+		return errors.Wrap(err, "roundChange invalid")
```

Comment:
just to understand, it was a bug that we validated only the inner message?

Reply:
I think so

Reply:
if so, we have to add a test for `message signers is empty`, `non unique signer` & `signer ID 0 not allowed`

Reply:
Yup, adding these tests.
Btw, the previous version couldn't produce any errors because, even though it should be `signedMsg.Validate()` and not `signedMsg.Message.Validate()`, the other checks were also verifying the conditions. But, of course, still better to call `signedMsg.Validate()`

Reply:
It is not a bug, but all other base validation functions for the other message types call validate on the upper message, and this is the only inconsistent one. Even though, all of these calls are indeed duplicated 😅